cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -70,6 +71,12 @@ public void getInstancePrefs(HttpRequest request, HttpResponder responder) {
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(preferencesService.getProperties()));
   }
 
+  @Path("/preferences/metadata")
+  @GET
+  public void getInstancePrefsMetadata(HttpRequest request, HttpResponder responder) throws Exception {
```

Comment:
Since this is a already a preference handler, just call the method as `getInstanceMetadata`.